okio/src/commonMain/kotlin/okio/internal/Buffer.kt

```diff
@@ -539,6 +483,34 @@ internal inline fun Buffer.commonWriteDecimalLong(v: Long): Buffer {
   return this
 }
 
+private fun countDigitsIn(v: Long): Int {
+  val guess = ((64 - v.countLeadingZeroBits()) * 10) ushr 5
+  return guess + (if (v > DigitCountToLargestValue[guess]) 1 else 0)
+}
+
+private val DigitCountToLargestValue = longArrayOf(
+  -1, // Every value has more than 0 digits.
+  9L, // For 1 digit (index 1), the largest value is 9.
+  99L,
+  999L,
+  9999L,
+  99999L,
+  999999L,
+  9999999L,
+  99999999L,
+  999999999L,
+  9999999999L,
+  99999999999L,
+  999999999999L,
+  9999999999999L,
+  99999999999999L,
+  999999999999999L,
+  9999999999999999L,
+  99999999999999999L,
+  999999999999999999L, // For 18 digits (index 18), the largest value is 999999999999999999.
+  Long.MAX_VALUE, // For 19 digits (index 19), the largest value is MAX_VALUE.
+)
```

Comment:
This is a slight deviation from the original table proposed in the article:

```kotlin
private val PowersOfTen = longArrayOf(
    0,
    10,
    100,
    1000,
    10000,
    100000,
    1000000,
    10000000,
    100000000,
    1000000000,
    10000000000,
    100000000000,
    1000000000000,
    10000000000000,
    100000000000000,
    1000000000000000,
    10000000000000000,
    100000000000000000,
    1000000000000000000
)
```

@swankjesse & I discovered that the original table wouldn't work for the `Long.MAX_VALUE` test case, as it has more digits than 10^18, and 10^19 is outside of the range of Long. The code instead compares against largest numbers that can be represented with a given digit count, which allowed us to fit another entry into the table!

Reply:
I saw that and it's a great improvement!